Bluetooth: Object transfer service object ID mask define

asbjornsabo opened the pull request:
Object IDs are 48 bits, but are often carried in uint64_t variables.
This PR defines a mask value that retains the least significant 48 bits.

The PR also adds defines for the maximum and minimum allowed values for object IDs, and moves the define for the directory listing object ID to the public header file.  This will allow users of the module to check that object ID values are valid.

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>

asbjornsabo committed:
Bluetooth: Object transfer service object ID mask define

Object IDs are 48 bits, but are often carried in uint64_t variables.
This commit defines a mask value that retains the least significant 48
bits.

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>

Thalley commented on include/bluetooth/services/ots.h:
```diff
@@ -31,6 +31,9 @@ extern "C" {
 #define BT_OTS_OBJ_ID_SIZE 6
 
+/** @brief Mask for OTS object IDs, preserving the 48 bits */
+#define BT_OTS_OBJ_ID_MASK 0x0000FFFFFFFFFFFFUL
```
I wonder if it would be more clear if it was 
```suggestion
#define BT_OTS_OBJ_ID_MASK BIT_MASK(48)
```

asbjornsabo replied:
Certainly easier to read than sitting there counting four zeros and twelve "F"s.
Changed.

asbjornsabo committed:
Bluetooth: Object Transfer Service: Defines for allowed object ID values

Add #defines for the maximum and minimum allowed values for object
IDs.

Moves the #define for the directory listing object ID to the public
header file

Having these limits available is useful for applications wanting to
ensure they pass in (or handle) valid object ID values.

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>
